core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java

```diff
@@ -74,7 +75,7 @@ public void checkParentPermission(Mode.Bits bits, LockedInodePath inodePath)
 
     // collects existing inodes info on the path. Note that, not all the components of the path have
     // corresponding inodes.
-    List<Inode> inodeList = inodePath.getInodeList();
+    List<InodeView> inodeList = (List<InodeView>) (List<?>) inodePath.getInodeList();
```

Comment:
This looks like a revert of #8307. Is it intentional?

Reply:
ping?

Reply:
Honestly can't remember why did I do that. Reverted.